rust/datafusion/src/execution/aggregate.rs

```diff
@@ -80,16 +80,37 @@ enum GroupByScalar {
     Utf8(String),
 }
 
-/// Common trait for all aggregation functions
+/// Aggregate function that can accept individual values and compute an aggregate
 trait AggregateFunction {
     /// Get the function name (used for debugging)
     fn name(&self) -> &str;
-    fn accumulate_scalar(&mut self, value: &Option<ScalarValue>) -> Result<()>;
-    fn result(&self) -> &Option<ScalarValue>;
+
+    /// Update the current aggregate value based on a new value. If rollup is false, then
```

Comment:
nit: can we also explain a little bit how the parameter `value` is used when it is set, and what's the behavior when it is not set? it will make it easier for people to maintain this code.